### ISSUE TITLE
add timer support

### ISSUE THREAD

**Issue by micah** [camptocamp/puppet-systemd#118]
systemd provides a cron-like system replacement, through the use of unit configuration files that end in ".timer". These unit configurations encode information about timers controlled and supervised by systemd, for timer-based activation.

It would be excellent if this module provided a mechanism for managing these!

**Comment by bastelfreak** [comment]
Hi,
this already works fine:
https://github.com/voxpupuli/puppet-borg/blob/f2c66b51c46d57b25326524415c0e9ced683a469/manifests/service.pp#L11-L26

https://github.com/voxpupuli/puppet-borg/blob/f2c66b51c46d57b25326524415c0e9ced683a469/templates/borg-backup.timer.epp#L1-L13

**Comment by trevor-vaughan** [comment]
@bastelfreak I agree that it works, but it's confusing for end users. It would be nice to have a defined type or something that just wraps it to be more clear.

**Comment by traylenator** [comment]
I was just contemplating writing  a systemd::timer  type , will try something.

**Comment by traylenator** [comment]
The only interface that makes sense would be

```puppet
systemd::timer{'runoften.timer':
    timer_content   => ...
    service_content =>
    enable => true,
    active => true,
}
```

Essentially to specify the content of the service and timer unit.

It does not make sense to specify the individual say `OnCalandar` arguments since this gets tricky because you then have to define the Before, , After, multiple OnCalendar, ...

The alternative is to only support really trivial jobs. Such as 

```puppet
systemd::timer{'myjob':
  exec => 'rm -rf /',
  on_calendar => 'daily',
}
```

along with some other timer options. Anything more complicated would require to resort back to two `systemd::unit_files` as per the above examples.

So is the latter one still worth while.

**Comment by trevor-vaughan** [comment]
@traylenator I think that it is. I also think that a wrapper that creates the rest of the bits for a generally useful `OnCalendar` example would be useful but could be a later addition.

**Comment by raphink** [comment]
Implemented in #138
